ouroboros-consensus/src/Ouroboros/Storage/ImmutableDB/Impl/Index/Cache.hs

```diff
@@ -161,7 +162,7 @@ data Cached hash = Cached
     -- from the current epoch, so it is worth optimising this case.
     -- Additionally, by appending to the current epoch through the cache, we
     -- are sure the current epoch info is never stale.
```

Comment:
```suggestion
    -- are sure the current epoch info is never stale.
    --
    -- We use an 'IntPSQ' here, where the keys are in fact epoch numbers. Since
    -- epoch numbers are internally represented by a 'Word64', one might be worried
    -- about a potential overflow here. While possible, it's not worth worrying about:
    -- - Whilst guaranteed to be only at least 30 bits, in practice, 64-bit GHC has 64-bit
    --   integers, so the conversion is bijective.
    -- - An epoch currently lasts around a week. Systems using a smaller representation
    --   might need to worry in a million years or so.
    -- - In the event of running for a million years, we're unlikely to have a problem anyway,
    --   since we only really cache _recent_ epochs. So the fact that they clash with the
    --   epochs from a million years ago isn't likely to be an issue.
```